### DISCUSSION TITLE
[Feature]: Reverse proxy documentation and examples (NPM etc.)

### DISCUSSION THREAD

**Discussion by ollioddi** [immich-app/immich#1593] [Feature Request]
### Feature detail

Hey.

I would like to put Immich behind a reverse proxy (In my case Nginx Proxy Manager), but find it a little bit hard to figure out how to.

Does anyone have examples for configuration with docker compose?

I found this thread on [reddit](https://www.reddit.com/r/selfhosted/comments/wp1zxd/immich_and_nginx_proxy_manager/) with various tips.

It seems like you shouldn't disable the immich docker network, since it uses it internally but i would like to know the following:

- How can i set a static IP on a custom VLAN
- How should NPM handle the different /api endpoint

I am using Unraid with docker-compose and have Immich running and several Docker networks with different VLANS.

A suggestion would be to add examples to the docs with various reverse proxies.

Let me know if this doesn't fit as a feature request.

Thanks in advance!

### Platform

Server

**Comment by jrasm91** [comment]
Correct, we even have a section (placeholder) for it already: "Guides".

For npm, which is what I use, I just update the immich-proxy container to also use the docker network that the npm container uses. Then the IP/hostname you use for the upstream server is `immich-proxy:8080`

If all the immich containers are on the same network, then the only "externally available" ip/port you'd need would be proxy container + port 2283. If you hook up the proxy container to the npm network you don't even need that, as you can use the internal service name + port instead.

![image](https://user-images.githubusercontent.com/4334196/216461305-90dd7c39-3a28-46e3-9b6f-7d875f4a8394.png)

**Comment by nep2ner** [comment]
How should one go about implementing this part of [the docs](https://immich.app/docs/administration/reverse-proxy) with NPM?

> All reverse proxies between Immich and the user must forward all headers and set the Host, X-Forwarded-Host, X-Forwarded-Proto and X-Forwarded-For headers to their appropriate values. Additionally, your reverse proxy should allow for big enough uploads. By following these practices, you ensure that all custom reverse proxies are fully compatible with Immich.

**Reply by notflip** [reply]
I'm wondering the same thing.

**Reply by RyGuy994** [reply]
anyone got this working?

**Reply by notflip** [reply]
I do! I got it working with nginx proxy manager. @RyGuy994

**Reply by RyGuy994** [reply]
@notflip do you have a guide or something I can follow?

**Reply by notflip** [reply]
@RyGuy994 I sent you an e-mail with 2 loom videos, I'm not sharing it here, I quickly recorded these while working, I didn't have time to blur out my url's etc.

**Reply by cryptedx** [reply]
@notflip 

Hey, would you be so kind to share it also with me. I am having a hard time to get it working.

Many thanks.

**Reply by reyfuu** [reply]
@notflip hey can i have that video too because iam still strugle to setup reverse proxy

**Reply by notflip** [reply]
@reyfuu @cryptedx Let me see if I still have it, otherwise I'll make a YT video.

**Reply by notflip** [reply]
@reyfuu @cryptedx See my video: https://youtu.be/_8SZIwP0cOY

**Reply by atselvan** [reply]
When I use the reverse-proxy url I keep getting below errors on the IOS app which blocks the backups:

```
Error deleting file path for iOS: PathNotFoundException: Cannot delete file, path = '/private/var/mobile/Containers/Data/Application/C6DBEE3F-6A75-4D8D-B97A-79F323C8C94F/tmp/.image/E5DCBB55-017D-4F9F-8FB3-50C2059CFC17_L0_001_1723614616.303693_o_DSC_0404.jpg' (OS Error: No such file or directory, errno = 2)
```

My nginx configuration:

```
client_max_body_size 50000M;
proxy_buffering off;
proxy_read_timeout 600s;
proxy_send_timeout 600s;
send_timeout 600s;
```

PS: I do not see these errors when I point to the IP.

**Comment by bo0tzz** [comment]
https://immich.app/docs/administration/reverse-proxy
https://immich.app/docs/guides/remote-access/

**Comment by chpalex** [comment]
I figured it out
![NPM Immich config](https://github.com/user-attachments/assets/b8ff0724-baf6-479f-ba07-e41cc134361d)
